File: apps/node/src/fizzbuzz.ts
export class FizzBuzz {
  generate(number: number): string {
    if (number % 15 === 0) {
      return "FizzBuzz";
    }
    if (number % 3 === 0) {
      return "Fizz";
    }
    if (number % 5 === 0) {
      return "Buzz";
    }

    return number.toString();
  }

  generateList(count: number): string[] {
    return Array.from({ length: count }, (_, index) =>
      this.generate(index + 1),
    );
  }

  printFizzBuzz(count: number): void {
    this.generateList(count).forEach((value) => {
      console.log(value);
    });
  }
}

File: apps/ruby/lib/fizz_buzz.rb
# frozen_string_literal: true

class FizzBuzz
  def self.generate(number)
    return 'FizzBuzz' if (number % 15).zero?
    return 'Fizz' if (number % 3).zero?
    return 'Buzz' if (number % 5).zero?

    number.to_s
  end

  def self.generate_list
    (1..100).map { |n| generate(n) }
  end
end
